fe/src/pages/campaigns/[id].vue

```diff
@@ -40,20 +47,81 @@ import { storeToRefs } from "pinia";
 import { toast } from "vue3-toastify";
 import { ethers } from "ethers";
 import CampaignProps from "~/types/CampaignProps";
-import CampaignSample from "~/mocks/campaign-sample.json";
 import { useWalletStore } from "~/store/wallet";
 const route = useRoute();
-const { truncate } = useUtils();
+const { truncate, getDaysLeft } = useUtils();
 
 const { $getSmartContract: getSmartContract } = useNuxtApp();
 
-const campaignValueSample = ref<CampaignProps>(CampaignSample);
-
 const isLoading = ref<boolean>(false);
-
+const canEdit = ref<boolean>(false);
 const useWallet = useWalletStore();
-const { isConnected } = storeToRefs(useWallet);
+const { isConnected, address } = storeToRefs(useWallet);
 const { id } = route.params;
+const campaign = ref<CampaignProps>({
+  campaignId: 0,
+  title: "",
+  creator: {
+    address: "",
+    fullName: "",
+    imgSrc: "",
+  },
+  imgSrc: "",
+  story: "",
+  daysLeft: 0,
+  totalDonation: 0,
+  totalBackers: 0,
+  donations: [],
+});
+
+const getCampaign = async (id: number): Promise<void> => {
+  isLoading.value = true;
+  const smartContract = await getSmartContract();
+  if (smartContract !== null) {
+    smartContract
+      .getCampaign(id)
+      .then((result) => {
+        campaign.value = {
+          campaignId: Number(result[0][0]),
+          title: result[0][3],
+          creator: {
+            address: result[0][1],
+            fullName: result[0][2],
+            imgSrc:
+              "https://img.freepik.com/free-psd/3d-illustration-person_23-2149436182.jpg",
+          },
+          imgSrc:
+            "https://images.unsplash.com/photo-1529390079861-591de354faf5",
+          story: result[0][4],
+          daysLeft: getDaysLeft(result[0][7]),
+          totalDonation: Number(ethers.formatEther(result[0][6])),
+          totalBackers: Number(result[0][8]),
+          donations: [],
+        };
```

Comment:
is there a way we can construct this in BE?

Reply:
@chevyB the data from BE looks like this:
- `result[0]` is for the Campaign details
- `result[1]` is the array of Donations of the Campaign

![image](https://github.com/framgia/sph-solidity/assets/57890826/ea6205e6-a4b6-47f7-abc2-a6ea3421635e)